File: app/src/main/java/com/example/filmproductionhelper/MaterialList.java
package com.example.filmproductionhelper;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

//list code learned from: https://www.youtube.com/watch?v=YmRPIGFftp0

public class MaterialList extends AppCompatActivity implements View.OnClickListener, AdapterView.OnItemClickListener {

    //attach items from xml
    private EditText itemET;
    private Button btn;
    private ListView itemsList;

    //working w/ lists
    private ArrayList<String> items; 
    private ArrayAdapter<String> adapter; //like an updater

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_material_list);

        itemET= findViewById(R.id.item_edit_text);
        btn= findViewById(R.id.add_b);
        itemsList= findViewById(R.id.items_list);

        items = FileHelper.readData(this);

        adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, items);
        itemsList.setAdapter(adapter);
        
        btn.setOnClickListener(this);
        itemsList.setOnItemClickListener(this);
    }

    //Add item in list when ADD button clicked on
    @Override
    public void onClick(View v) {
        switch(v.getId()){
            case R.id.add_b:
                String itemEntered= itemET.getText().toString(); //gets text entered
                adapter.add(itemEntered); // sets/adapts it as new item
                itemET.setText("");

                FileHelper.writeData(items, this);
                Toast.makeText(this, "Item Added", Toast.LENGTH_SHORT).show();
                break;
        }
    }

    //Delete when item clicked on
    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        items.remove(position);
        adapter.notifyDataSetChanged();
        FileHelper.writeData(items, this);
        Toast.makeText(this, "Delete", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/filmproductionhelper/MaterialList.kt
package com.example.filmproductionhelper

import android.content.DialogInterface
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.ImageView
import android.widget.PopupMenu
import android.widget.TextView
import com.example.filmproductionhelper.DTO.ToDo
import kotlinx.android.synthetic.main.activity_material_list.*

//code learned from: https://www.youtube.com/watch?v=OqRO-B-zlKU

class MaterialList : AppCompatActivity() {

    lateinit var dbHandler: DBHandler //database handler for SQLite

    override fun onCreate(savedInstanceState: Bundle?) {
        title = "Material Lists" //setting up layout
        dbHandler = DBHandler(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_material_list)
        rv_dashboard.layoutManager = LinearLayoutManager(this)
        setSupportActionBar(dashboard_toolbar)

        fab_dashboard.setOnClickListener {//clicking on the Plus button
            val dialog = AlertDialog.Builder(this)
            val view = layoutInflater.inflate(R.layout.lists_popup, null)
            val toDoName = view.findViewById<EditText>(R.id.lists_insert)
            dialog.setTitle("Add List")
            dialog.setView(view)

            dialog.setPositiveButton("Create List") { _: DialogInterface, _: Int -> //Create List Button
                if (toDoName.text.isNotEmpty()) {
                    val toDo = ToDo()
                    toDo.name = toDoName.text.toString()
                    dbHandler.addToDo(toDo)
                    refreshList()
                }
            }
            dialog.setNegativeButton("Cancel") { _: DialogInterface, _: Int -> } //Cancel Button
            dialog.show()
        }
    } //Create Lists

    fun updateLists(toDo: ToDo){
        val dialog = AlertDialog.Builder(this)
        dialog.setTitle("Update List Name")
        val view = layoutInflater.inflate(R.layout.lists_popup, null)
        val toDoName = view.findViewById<EditText>(R.id.lists_insert)
        toDoName.setText(toDo.name)
        dialog.setView(view)
        dialog.setPositiveButton("Update") { _: DialogInterface, _: Int ->
            if (toDoName.text.isNotEmpty()) {

                toDo.name = toDoName.text.toString()
                dbHandler.updateToDo(toDo)
                refreshList()
            }
        }
        dialog.setNegativeButton("Cancel") { _: DialogInterface, _: Int -> }
        dialog.show()
    } //Update List Name

    override fun onResume() {
        refreshList()
        super.onResume()
    }

    private fun refreshList(){
        rv_dashboard.adapter = DashboardAdapter(this, dbHandler.getToDos())
    }

    class DashboardAdapter(val activity: MaterialList, val list: MutableList<ToDo>) :
            RecyclerView.Adapter<DashboardAdapter.ViewHolder>() {
        override fun onCreateViewHolder(p0: ViewGroup, p1: Int): ViewHolder {
            return ViewHolder(LayoutInflater.from(activity).inflate(R.layout.rv_child_dashboard, p0, false))
        }

        override fun getItemCount(): Int {
            return list.size
        }

        override fun onBindViewHolder(holder: ViewHolder, p1: Int) {
            holder.toDoName.text = list[p1].name

            holder.toDoName.setOnClickListener{
                val intent = Intent(activity, ItemActivity::class.java)
                intent.putExtra(INTENT_TODO_ID, list[p1].id)
                intent.putExtra(INTENT_TODO_NAME,list[p1].name)
                activity.startActivity(intent)
            }
            holder.menu.setOnClickListener {
                val popup = PopupMenu(activity,holder.menu)
                popup.inflate(R.menu.dashboard_child)
                popup.setOnMenuItemClickListener {

                    when(it.itemId){
                        R.id.menu_edit->{
                            activity.updateLists(list[p1])
                        }
                        R.id.menu_delete->{
                            activity.dbHandler.deleteToDo(list[p1].id)
                            activity.refreshList()
                        }
                        R.id.menu_mark_as_completed->{
                            activity.dbHandler.updateToDoItemCompletedStatus(list[p1].id,true)
                        }
                        R.id.menu_reset->{
                           activity.dbHandler.updateToDoItemCompletedStatus(list[p1].id,false)
                        }
                    }

                    true
                }
                popup.show()
            }
        } //create lists sub-materials

        class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
            val toDoName: TextView = v.findViewById(R.id.tv_todo_name)
            val menu : ImageView = v.findViewById(R.id.iv_menu)
        }
    } //Adapter for List/Recycler View
}
